packages/smooth_app/lib/pages/user_preferences_page.dart

```diff
@@ -227,11 +218,389 @@ class UserPreferencesPage extends StatelessWidget {
+    final ThemeProvider themeProvider,
+  ) =>
+      TextButton(
+        onPressed: () async => themeProvider.setColorTag(colorTag),
+        style: TextButton.styleFrom(
+          backgroundColor: SmoothTheme.getColor(
+            colorScheme,
+            SmoothTheme.MATERIAL_COLORS[colorTag]!,
+            ColorDestination.BUTTON_BACKGROUND,
+          ),
+        ),
+        child: Icon(
+          Icons.palette,
+          color: SmoothTheme.getColor(
+            colorScheme,
+            SmoothTheme.MATERIAL_COLORS[colorTag]!,
+            ColorDestination.BUTTON_FOREGROUND,
+          ),
+        ),
+      );
+
+  String _getPreferenceFlagKey(final int groupIndex) {
+    switch (groupIndex) {
+      case 0:
+        return 'profile';
+      case 1:
+        return 'food';
+      case 2:
+        return 'vrac';
+    }
+    throw Exception('unknown group index: $groupIndex');
+  }
+
+  bool _isCollapsed(
+    final int groupIndex,
+    final UserPreferences userPreferences,
+  ) =>
+      userPreferences.getFlag(_getPreferenceFlagKey(groupIndex)) ??
+      // only group "food" is supposed to be expanded by default at init time
+      groupIndex != 1;
+
+  Future<void> _switchCollapsed(
+    final int groupIndex,
+    final UserPreferences userPreferences,
+  ) async =>
+      userPreferences.setFlag(
+        _getPreferenceFlagKey(groupIndex),
+        !_isCollapsed(
+          groupIndex,
+          userPreferences,
+        ),
+      );
+
+  Widget _getGroupTitle(
+    final int groupIndex,
+    final UserPreferences userPreferences,
+    final ThemeData themeData,
+  ) {
+    final String label;
+    final String subLabel;
+    final bool extra = groupIndex == 0;
+    switch (groupIndex) {
+      case 0:
+        label = 'Your Profile';
+        subLabel = 'Set app settings and find out advices and blah blah';
+        break;
+      case 1:
+        label = 'Food Preferences';
+        subLabel = 'Choose what information about food matters most to you';
+        break;
+      case 2:
+        label = 'App Settings';
+        subLabel = 'Dark mode, country, color, ...';
+        break;
+      default:
+        throw Exception('unknown group index: $groupIndex');
+    }
+    Widget title = Text(label, style: themeData.textTheme.headline2);
+    if (extra) {
+      title = Row(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          title,
+          Container(
+            width: 15,
+            height: 15,
+            child: Center(
+              child: Text(
+                '1',
+                style: TextStyle(
+                  color: Colors.white,
+                  fontSize: 8,
+                  fontWeight: FontWeight.bold,
+                ),
+              ),
+            ),
+            decoration: BoxDecoration(
+              color: Colors.red,
+              shape: BoxShape.circle,
+            ),
+          ),
+        ],
+      );
```

Comment:
I don't think we need this red dot from the mocks, it does look good, but only a hardcoded one is confusing. We currently have nothing to display with this so we don't need the additional code.

Reply:
Fair enough, it was just for the screenshot. Removing it...